src/bootstrap/check.rs

```diff
@@ -115,7 +107,37 @@ pub fn cargo(build: &Build, stage: u32, host: &str) {
     // available.
     cargo.env("CFG_DISABLE_CROSS_TESTS", "1");
 
-    build.run(cargo.env("PATH", newpath));
+    build.run(cargo.env("PATH", &path_for_cargo(build, compiler)));
+}
+
+/// Runs `cargo test` for the rls.
+pub fn rls(build: &Build, stage: u32, host: &str) {
+    let compiler = &Compiler::new(stage, host);
+
+    let mut cargo = build.cargo(compiler, Mode::Tool, host, "test");
+    cargo.arg("--manifest-path").arg(build.src.join("src/tools/rls/Cargo.toml"));
+
+    // Don't build tests dynamically, just a pain to work with
+    cargo.env("RUSTC_NO_PREFER_DYNAMIC", "1");
+
+    let mut dylib_path = dylib_path();
+    dylib_path.insert(0, build.sysroot_libdir(&compiler, host));
+    cargo.env(dylib_path_var(), env::join_paths(&dylib_path).unwrap());
+
+    // Avoid using the rustc shim.
+    cargo.env("RUSTC", build.compiler_path(compiler));
+
+    build.run(cargo.env("PATH", &path_for_cargo(build, compiler)));
```

Comment:
The RLS does not do this. However, if I don't do this PATH hack, I get an error about different versions of the bitflags crate. Possibly cargo culting and I need to change something else?